### PR TITLE
[C++/it] italian translation of C++

### PR DESCRIPTION
**Pull request by sinkswim**
Ciao! Added italian translation of C++ guide

### CODE REVIEW

**Comment by geoffliu** [comment]
@sinkswim thanks for the translation! I'll have some time this weekend to look over this. In the meantime, I added two new sections to the English version, maybe you would like to update the translation? :)

**Comment by sinkswim** [comment]
@geoffliu I've added the Template and Fun Stuff section. Now my version is up to date!
Cheers!

File: it-it/c++-it.html.markdown [hunk 1]
```diff
@@ -0,0 +1,720 @@
...
+// Proprio come nel C, l'inizio del programma è una funzione chiamata
+// main con un intero come tipo di ritorno,
+// Questo valore serve come stato d'uscita del programma.
+// Vedi http://it.wikipedia.org/wiki/Valore_di_uscita per maggiori informazioni.
+int main(int argc, char** argv)
+{
+    // Gli argomenti a linea di comando sono passati tramite argc e argv così come
+    // avviene in C.
+    // argc indica il numero di argomenti,
+    // e argv è un array in stile-C di stringhe (char*)
```
Comment (geoffliu) [inline it-it/c++-it.html.markdown]: un array di stringhe in stile-C

File: it-it/c++-it.html.markdown [hunk 2]
```diff
@@ -0,0 +1,720 @@
...
+    // che rappresenta gl iargomenti.
```
Comment (geoffliu) [inline it-it/c++-it.html.markdown]: gli argomenti

File: it-it/c++-it.html.markdown [hunk 3]
```diff
@@ -0,0 +1,720 @@
...
+
+///////////////
+// Input/Output
+///////////////
+
+// L'input e l'output in C++ utilizza gli streams
+// cin, cout, e cerr i quali rappresentano stdin, stdout, e stderr.
+// << è l'operatore di inserzione >> è l'operatore di estrazione.
+
+#include <iostream> // Include for I/O streams
```
Comment (geoffliu) [inline it-it/c++-it.html.markdown]: Missed a line here.

File: it-it/c++-it.html.markdown [hunk 4]
```diff
@@ -0,0 +1,720 @@
...
+    // stampa "Il cane è <nome> e pesa <peso>"
+    //        "Il cane è di <proprietario>"
+}
+
+///////////////////////////////////////////////////
+// Inizializzazione ed Overloading degli Operatori
+//////////////////////////////////////////////////
+
+// In C++ puoi sovrascrivere il comportamento di operatori come +, -, *, /, ecc...
+// Questo è possibile definendo una funzioneche viene chiamata
```
Comment (geoffliu) [inline it-it/c++-it.html.markdown]: funzione che

File: it-it/c++-it.html.markdown [hunk 5]
```diff
@@ -0,0 +1,720 @@
...
+    input.abbaia();
+}
+
+// Nota che niente è specificato relativamente al tipo di parametri. Il compilatore
+// genererà  e poi verificherà il tipo di ogni invocazione del template, così che
+// la funzione di cui sopra funzione con ogni tipo 'T' che ha const 'abbaia' come metodo!
+
+Cane fluffy;
+fluffy.impostaNome("Fluffy")
+abbaiaTreVolte(fluffy); // Stampa "Fluffy abbaia tre volte.
```
Comment (geoffliu) [inline it-it/c++-it.html.markdown]: Stampa "Fluffy abbaia" tre volte.

File: it-it/c++-it.html.markdown [hunk 6]
```diff
@@ -0,0 +1,720 @@
...
+//   vettori (i.e. array auto-aggiustati), mappe hash, e così via
+//   sono tutti automaticamente distrutti con i loro contenuti quando escono dalla visibilità.
+// - I mutex usano lock_guard e unique_lock
+
+///////////////////////
+// Roba divertente
+//////////////////////
+
+// Aspetti del C++ che potrebbero sbalordire i nuovi arrivati (e anche qualche veterano).
+// Questa sezione è, fortunatamente, selvaggiamente incompleta; il C++ è uno dei linguaggi
```
Comment (geoffliu) [inline it-it/c++-it.html.markdown]: sfortunatamente

**Comment by sinkswim** [comment]
Woo, typo parade. Fixed. Thank you!

**Comment by geoffliu** [comment]
Cool, thanks!

**Pull request merged**